### DISCUSSION TITLE
cron job running against cached postgress db data

### DISCUSSION THREAD

**Discussion by robertnowell** [vercel/community#6974] [Help] [answered]
### Summary

I have a simple cron job that queries my postgres database, sends some emails, and then updates the same db

every time the cron runs, it is running against stale db data, even though the db has been updated since then

web app is hitting the same db but is getting fresh data every time -- not hitting the cache

after manually clearing the data cache, the cron will get fresh data the first time, and every subsequent run will return the same data

so it seems that the data cache is returning stale data from the db (even though the cron logs show cache hit: false for the api call, so it seems like the cache hit is on the call to read from the db, not the cron api call, but this is not logged as far as I can see)

curious if this an issue with the settings of my data cache, my db, or another issue. 

any steps other than disabling data cache recommended?

### Example

cron job is sending the same emails every time it runs on the /summarize endpoint (because it isn't detecting they've already been sent)

but when I send a request to /summarize locally, it sends the correct up to date emails

and when I access the web app at thoughtpartner.vercel.app it retrieves up to date data from the same postgres db

### Steps to Reproduce

I have a cron with a simple route and a vercel postrgres db

**Comment by versecafe** [comment, marked as answer]
@robertnowell Next 14 Caches GET requests by default, this is being fixed in 15 but for now unless you want to bump to the 15rc you can always force the route to dynamic with no caching like so

```ts
export const dynamic = "force-dynamic";
export const fetchCache = "force-no-store";
```

**Reply by robertnowell** [reply]
thank you @versecafe! super helpful -- this worked

**Reply by versecafe** [reply]
The caching behaviour in Next 14 with the app router tends to cause trouble for a lot of people, glad it was a simple fix.
